src/frontend/src/tests/lib/components/transactions/AllTransactionsList.spec.ts

```diff
@@ -1,28 +1,72 @@
-import { btcTransactionsStore } from '$btc/stores/btc-transactions.store';
-import {
-	BTC_MAINNET_TOKEN_ID,
-	BTC_REGTEST_TOKEN_ID,
-	BTC_TESTNET_TOKEN_ID
-} from '$env/tokens.btc.env';
-import { ethTransactionsStore } from '$eth/stores/eth-transactions.store';
+import BtcTransaction from '$btc/components/transactions/BtcTransaction.svelte';
 import AllTransactionsList from '$lib/components/transactions/AllTransactionsList.svelte';
+import { mapAllTransactionsUi } from '$lib/utils/transactions.utils';
+import { createMockBtcTransactionsUi } from '$tests/mocks/btc.mock';
 import en from '$tests/mocks/i18n.mock';
 import { render } from '@testing-library/svelte';
+import type { MockedFunction } from 'vitest';
+
+vi.mock('$lib/utils/transactions.utils', () => ({
```

Comment:
why mocking the utils?

Reply:
I want to assume that the util function works correctly, and I just verify how the component behaves with different outputs of the util. I don't think it is necessary to test the function inside the component, since the component behavior changes only in two cases: either empty return or non-empty return, right?

Reply:
Yeah but you also want to test the rendering so per extension you need the utils, no?

Reply:
yes yes, but i mock the util return, for example, I mock an empty return or a return with n transactions, to test the rendering in both cases; is that what you meant? or do you mean that it should render something else?

Reply:
I just meant: don't mock the utils 😁

If possible. If we introduce various level of mocks within the all code base, then the tests become less accurate and might actually miss a case between mocked layers. That's why when it's not too difficult, kind of feel like it's not too bad to not mock, if that makes sense.

Reply:
Ah understood now! Done! have a look please